sycl/ReleaseNotes.md

```diff
@@ -16,6 +16,699 @@
+
+### Sanitizers
+
+#### Memory Sanitizer
+
+- Introduced memory sanitizer support. intel/llvm#15955, intel/llvm#16427,
+  intel/llvm#16478, intel/llvm#16935, intel/llvm#16535, intel/llvm#16477,
+  intel/llvm#16567, intel/llvm#16526, intel/llvm#16678, intel/llvm#16566,
+  intel/llvm#16619, intel/llvm#16705
+
+  It features:
+  - Checking for uses of uninitalized values in private memory. intel/llvm#17309
+  - Checking for uses of unitialized values in local memory, such
+    as `local_accessor` or `group_local_memory`. intel/llvm#17180,
+    intel/llvm#17054
+  - Sanitizing USM operations like `memset` or `memcpy`. intel/llvm#16511
+
+#### Thread Sanitizer
+
+- Introduced thread sanitizer support for device code. intel/llvm#17345,
+  intel/llvm#17211, intel/llvm#17155, intel/llvm#17181
+
+## Improvements and bugfixes
+
+### `sycl_ext_oneapi_graph` extension
+
+- Reimplemented topological sort algorithm used to determine graph nodes
+  execution order to avoid issues with overflowing stack on huge graphs and
+  improve performance. intel/llvm#17495
+- Documented kernel binary update feature which allows to update kernel nodes
+  in graphs. This feature had been implemented earlier already. intel/llvm#14896
+- Introduced ability to update host-task nodes in graphs. intel/llvm#16853
+- Fixed race condition in `mutable_command_graph` node queries. intel/llvm#17012
+- Fixed the issue with not all graph-related classes fully implementing
+  common reference semantics. intel/llvm#16788
+- Documented interaction with `sycl_ext_oneapi_local_memory` extension.
+  intel/llvm#16379
+- Documented interaction with `sycl_ext_oneapi_work_group_memory` extension.
+  intel/llvm#16229
+- Made `ext_oneapi_weak_object` extension work with graph objects.
+  intel/llvm#16209
+- Fixed a bug where using `local_accessor` or `work_group_memory` objects as
+  part of whole graph update would function incorrectly on CUDA & HIP backends.
+  intel/llvm#16025
+
+### SYCLcompat library
+
+- Introduced new set of group utility functions and classes aimed to reduce the
+  gap between `syclcompat` and `dpct` namespaces. intel/llvm#17263
+- 73e6b224aacf [SYCLCOMPAT] Forward launch arguments to avoid copies (#16965)
+  - Definitely user-visible, but I'm not sure how to word that
+- Fixed `compare_mask` putting results in the wrong 2-byte segment of 4-byte
+  output. intel/llvm#16768
+- Optimized implementation of `permute_sub_group_by_xor` for the case when
+  `logical_sub_group_size == 32`. intel/llvm#16646
+- Added new function `ternary_logic_op` to perform bitwise logical operations
+  on three input values based on the specified 8-bit truth table.
+  intel/llvm#16509
+- 6e0d90e73ed1 [SYCLCompat] Fix vectorized_binary impl to make SYCLomatic migrated code run pass (#16553)
+  - Not sure how to word that
+- 16c447998836 [SYCL][COMPAT] Replace T{-1} with static_cast<T>(-1) for mask creation (#16527)
+  - bugfix?
+
+### Explicit SIMD extension
```

Comment:
@intel/dpcpp-esimd-reviewers, tagging you for review

Reply:
@intel/dpcpp-esimd-reviewers, friendly ping